modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -106,5 +118,92 @@ object ConstructionService {
       } else {
         parseUnsignedTransaction(hex)
       }
+
+    def combineTransaction(hex: Hex, signatures: NonEmptyList[RosettaSignature]): EitherT[F, Throwable, Either[ConstructionError, Hex]] =
```

Comment:
these two methods written in a functional way could be dramatically simplified. also, method for converting signatures is overengineered and the traversing is obfuscated. it's recommended to use traversing provided by cats:
```scala
    def combineTransaction(hex: Hex, signatures: NonEmptyList[RosettaSignature]): EitherT[F, ConstructionError, Hex] =
      EitherT
        .fromEither(KryoSerializer[F].deserialize[Transaction](hex.toBytes))
        .leftMap(_ => MalformedTransaction)
        .flatMap { transaction =>
          convertSignatures(signatures).flatMap { signatures =>
            EitherT
              .fromEither(
                KryoSerializer[F]
                  .serialize(Signed[Transaction](transaction, NonEmptySet(signatures.head, SortedSet(signatures.tail: _*))))
              )
              .leftMap(_ => MalformedTransaction.asInstanceOf[ConstructionError])
              .map(Hex.fromBytes(_))
          }
        }

    def convertSignatures(signatures: NonEmptyList[RosettaSignature]): EitherT[F, ConstructionError, NonEmptyList[SignatureProof]] =
      EitherT {
        signatures.traverse { signature =>
          signature.publicKey.hexBytes.toPublicKeyByEC.map { pk =>
            signature.signatureType match {
              case SignatureType.Ecdsa => SignatureProof(pk.toId, Signature(signature.hexBytes)).asRight[ConstructionError]
              case _                   => UnsupportedOperation.asInstanceOf[ConstructionError].asLeft[SignatureProof]
            }
          }
        }.map(_.sequence)
      }

```

Reply:
So, use `traverse` instead of `foldLeft`?

Reply:
I've implemented the functions as shown, but I will need to study them more to understand them.

Reply:
The return type for `combineTransaction` is still `EitherT[F, Throwable, Either[ConstructionError, Hex]]`. I would like it to be `EitherT[F, ConstructionError, Hex]`. I think we did a `liftTo` in a previous, similar situation, but I don't completely understand lifting and `liftTo`. Am I looking in the right direction here?

Reply:
What do you mean that the return type is still `EitherT[F, Throwable, Either[ConstructionError, Hex]]`?
It's of type `EitherT[F, ConstructionError, Hex]`, you can see that clearly here:
![image](https://user-images.githubusercontent.com/4959931/224950825-54823fb6-3b16-4d04-9884-04369f042ecb.png)

Reply:
My mistake on the return type. The error was on my side.